File: ITapply/ITapply.Services/Database/Review.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using static ITapply.Services.Database.Enums;

namespace ITapply.Services.Database
{
    public class Review
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int CandidateId { get; set; }
        [ForeignKey("CandidateId")]
        public Candidate Candidate { get; set; }

        [Required]
        public int EmployerId { get; set; }
        [ForeignKey("EmployerId")]
        public Employer Employer { get; set; }

        [Required]
        [Range(1, 5)]
        public int Rating { get; set; }

        [Required]
        [StringLength(3000)]
        public string Comment { get; set; }

        [Required]
        public ReviewRelationship Relationship { get; set; }

        [Required]
        [StringLength(100)]
        public string Position { get; set; } // Position held by the reviewer at the company

        [Required]
        public ModerationStatus ModerationStatus { get; set; } = ModerationStatus.Pending; // Default status

        [Required]
        public DateTime ReviewDate { get; set; } = DateTime.UtcNow;
    }
}

File: ITapply/ITapply.UI/itapply_mobile/lib/models/review.dart
import 'package:itapply_desktop/models/enums.dart';
import 'package:json_annotation/json_annotation.dart';

part 'review.g.dart';
@JsonSerializable(explicitToJson: true)
class Review {
  final int id;
  final int candidateId;
  final String? candidateName;
  final int employerId;
  final String? companyName;
  final int rating;
  final String? comment;
  @JsonKey(unknownEnumValue: ReviewRelationship.formerEmployee)
  final ReviewRelationship relationship;
  final String? position;
  @JsonKey(unknownEnumValue: ModerationStatus.pending)
  final ModerationStatus moderationStatus;
  final DateTime reviewDate;

  Review({
    required this.id,
    required this.candidateId,
    this.candidateName,
    required this.employerId,
    this.companyName,
    required this.rating,
    this.comment,
    required this.relationship,
    this.position,
    required this.moderationStatus,
    required this.reviewDate,
  });

  factory Review.fromJson(Map<String, dynamic> json) => _$ReviewFromJson(json);
  Map<String, dynamic> toJson() => _$ReviewToJson(this);
}